changelogs/fragments/2824-gitlab_project-project-under-user.yml

```diff
@@ -0,0 +1,4 @@
+---
+minor_changes:
+  - gitlab_project - user projects are created using namespace ID, instead of user ID (https://github.com/ansible-collections/community.general/pull/2824).
+  - gitlab_project - projects can be created under other user's namespaces (https://github.com/ansible-collections/community.general/pull/2824).
```

Comment:
I'm not sure why this change needs two separate changelog fragments. What is the relation between these fragments to each other, and to this change (adding a new module option, which isn't mentioned in any of them)?

Reply:
@stgrace ^

Reply:
So first thing we did was try out the module. My colleague was trying the module out with a personal access key made under his account, and somehow projects were being created under another colleagues account. Turns out the user ID for my colleague was 11, and his workspace ID was 16. The user ID of the other colleague under which the projects were being created was 6, but her workspace ID was 11. 

TL;DR
The module was picking the user ID, but should have been picking the user's workspace ID. We fixed this, and added an option to create projects under other user's workspaces.

Reply:
In that case, one is a bugfix, while the other is a feature. These should be two separate PRs (because one can be backported to stable-2 and the other not), and the bugfix changelog fragment should be in the `bugfixes` section.

Anyway, I don't see how this actually changes something in the case `username` is not provided. `namespace` is set in line 374 the same way as before.

Reply:
The getting of the namespace is done with a different ID that is set in the changes we made. The variable name did not change but the value did. Before it was retrieved using UID, now it is using the user's workspace ID.

I can make different PR's for it

Reply:
Ah, I missed the `gitlab_instance.users.list` → `gitlab_instance.namespaces.list`. In any case, splitting this up in two PRs is a good idea (I'd create a new PR for the bugfix, and keep this one for the feature as a WIP until the bugfix PR is merged).

But shouldn't `user_group_id` be renamed to `namespace_id` then? And why not just keep the namespace, instead of retrieving it a second time shortly after?

Also I'm a bit worried on two things:
1. You are listing workspaces when searching for the user name, instead of listing users. Is that really the correct approach?
2. What happens to the `group_identifier` branch? There, `user_group_id` is set to `group.id`. Is this also a namespace ID?

Reply:
I'll make a new PR and continue the discussion there